src/cdomain/value/cdomains/intDomain.ml

```diff
@@ -2256,9 +2256,8 @@ struct
       shift_op a b
     in
     (* If one of the parameters of the shift is negative, the result is undefined *)
-    let x_min = minimal x in
-    let y_min = minimal y in
-    if x_min = None || y_min = None || Z.compare (Option.get x_min) Z.zero < 0 || Z.compare (Option.get y_min) Z.zero < 0 then
+    let is_negative = Option.fold ~none:true ~some:(fun x -> Z.compare x Z.zero < 0) in
```

Comment:
Any reason not to use the dedicated comparison operators from Z? I think these would make the code easier to use and would be a nice improvement to do together with this :)

Reply:
This fold could just be `GobOption.for_all` anyway.